app/dashboard/src/modals/InvitedToOrganizationModal.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, ButtonGroup, Dialog, Text } from '#/components/AriaComponents'
+import { backendMutationOptions } from '#/hooks/backendHooks'
+import { useAuth, useFullUserSession } from '#/providers/AuthProvider'
+import { useRemoteBackendStrict } from '#/providers/BackendProvider'
+import { useText } from '#/providers/TextProvider'
+import { useMutation } from '@tanstack/react-query'
+
+// ==================================
+// === InvitedToOrganizationModal ===
+// ==================================
+
+/** Modal for accepting the terms of service. */
+export function InvitedToOrganizationModal() {
+  const { getText } = useText()
+  const { session } = useAuth()
+  const backend = useRemoteBackendStrict()
+  const { user } = useFullUserSession()
+  const shouldDisplay = user.newOrganizationName != null && user.newOrganizationInvite != null
+
+  const acceptInvitation = useMutation(
+    backendMutationOptions(backend, 'acceptInvitation'),
+  ).mutateAsync
+  const declineInvitation = useMutation(
+    backendMutationOptions(backend, 'declineInvitation'),
+  ).mutateAsync
+
+  if (!shouldDisplay) {
+    return <Outlet context={session} />
+  } else {
+    switch (user.newOrganizationInvite) {
+      case 'pending': {
+        return (
+          <>
+            <Outlet context={session} />
+            <Dialog
+              title={getText('organizationInviteTitle')}
+              isKeyboardDismissDisabled
+              isDismissable={false}
+              hideCloseButton
+              modalProps={{ defaultOpen: true }}
+            >
+              {({ close }) => (
+                <div className="flex flex-col gap-4">
+                  <div>
+                    <Text disableLineHeightCompensation>{getText('organizationInvitePrefix')}</Text>
+                    <Text disableLineHeightCompensation className="font-bold">
+                      {user.newOrganizationName}
+                    </Text>
+                    <Text disableLineHeightCompensation>{getText('organizationInviteSuffix')}</Text>
+                  </div>
+                  <ButtonGroup className="w-min self-end">
+                    <Button
+                      variant="outline"
+                      onPress={async () => {
+                        await declineInvitation([user.email])
+                        close()
+                      }}
+                    >
+                      {getText('decline')}
+                    </Button>
+                    <Button
+                      variant="tertiary"
+                      onPress={async () => {
+                        await acceptInvitation([])
+                        close()
+                      }}
+                    >
+                      {getText('accept')}
+                    </Button>
+                  </ButtonGroup>
+                </div>
+              )}
+            </Dialog>
+          </>
+        )
+      }
+      case 'error': {
```

Comment:
Why do we have different dialogs, instead of using a single one?
Why the `acceptInvitation` doesn't return an error?

Reply:
not sure... just following what the backend is giving us. the `error` here is present on the user object

Reply:
Not sure it's optimal. The error should not appear if a user accepted got an error and reloaded the page. 
Unless he's either accepted or declined, he's supposed to see the invite dialog.

(Ideally, it should be a tooltip though)